# HG changeset patch
# User Torsten <mttl@mailbox.org>
# Date 1504367715 -7200
#      Sat Sep 02 17:55:15 2017 +0200
# Node ID a5b20fc6588a7d8545f3d9468a1b3cee2eed35a4
# Parent  d511f7e1565272dc88b4a23af97793818bf0868c
editor: more consistent behavior of lon line marker settings in (bug #51872)

* file-editor-tab.cc (notice_settings): new settings for long line marker,
  one flag for active or not and two other flags for line or background

* file-editor.cc (show_long_line, notice_settings): respect new settings flags

* settings-dialog.cc (settings_dialog): set the updated dialog objects to
  the related values of the settings file;
  (write_changed_settings): write the values form the dialog into the settings
  file

* settings-dialog.ui: changed objects for long line marker, a checkbox for
  enabling it and radio buttons for the implementation (line or background)

diff --git a/libgui/src/m-editor/file-editor-tab.cc b/libgui/src/m-editor/file-editor-tab.cc
--- a/libgui/src/m-editor/file-editor-tab.cc
+++ b/libgui/src/m-editor/file-editor-tab.cc
@@ -2414,21 +2414,32 @@ file_editor_tab::notice_settings (const 
   _long_title = settings->value ("editor/longWindowTitle", false).toBool ();
   update_window_title (_edit_area->isModified ());
 
   _auto_endif = settings->value ("editor/auto_endif",1).toInt ();
 
   // long line marker
   int line_length = settings->value ("editor/long_line_column",80).toInt ();
   _edit_area->setEdgeColumn (line_length);
-  _edit_area->setEdgeMode (QsciScintilla::EdgeNone);
+
   if (settings->value ("editor/long_line_marker",true).toBool ())
-    _edit_area->setEdgeMode (QsciScintilla::EdgeLine);
-  if (settings->value ("editor/long_line_marker_background",true).toBool ())
-    _edit_area->setEdgeMode (QsciScintilla::EdgeBackground);
+    {
+      if (settings->value ("editor/long_line_marker_line",true).toBool ())
+        _edit_area->setEdgeMode (QsciScintilla::EdgeLine);
+      else
+        {
+          if (settings->value ("editor/long_line_marker_background",false)
+                        .toBool ())
+            _edit_area->setEdgeMode (QsciScintilla::EdgeBackground);
+          else
+            _edit_area->setEdgeMode (QsciScintilla::EdgeLine);
+        }
+    }
+  else
+    _edit_area->setEdgeMode (QsciScintilla::EdgeNone);
 
   // line wrapping and breaking
   _edit_area->setWrapVisualFlags (QsciScintilla::WrapFlagByBorder);
   _edit_area->setWrapIndentMode (QsciScintilla::WrapIndentSame);
 
   if (settings->value ("editor/wrap_lines",false).toBool ())
     _edit_area->setWrapMode (QsciScintilla::WrapWord);
   else
diff --git a/libgui/src/settings-dialog.cc b/libgui/src/settings-dialog.cc
--- a/libgui/src/settings-dialog.cc
+++ b/libgui/src/settings-dialog.cc
@@ -413,22 +413,26 @@ settings_dialog::settings_dialog (QWidge
   _editor_current_line_color = new color_picker (setting_color);
   ui->editor_grid_current_line->addWidget (_editor_current_line_color,0,3);
   _editor_current_line_color->setMinimumSize (20,10);
   _editor_current_line_color->setEnabled (false);
   connect (ui->editor_highlightCurrentLine, SIGNAL (toggled (bool)),
            _editor_current_line_color, SLOT (setEnabled (bool)));
   ui->editor_highlightCurrentLine->setChecked (
     settings->value ("editor/highlightCurrentLine",true).toBool ());
-  ui->editor_long_line_marker_none->setChecked (
-    settings->value ("editor/long_line_marker_none",false).toBool ());
   ui->editor_long_line_marker->setChecked (
-    settings->value ("editor/long_line_marker",true).toBool ());
-  ui->editor_long_line_marker_background->setChecked (
-    settings->value ("editor/long_line_marker_background",false).toBool ());
+            settings->value ("editor/long_line_marker",true).toBool ());
+  bool long_line =
+        settings->value ("editor/long_line_marker_line",true).toBool ();
+  ui->editor_long_line_marker_line->setChecked (long_line);
+  bool long_back =
+        settings->value ("editor/long_line_marker_background",false).toBool ();
+  ui->editor_long_line_marker_background->setChecked (long_back);
+  if (! (long_line || long_back))
+    ui->editor_long_line_marker_line->setChecked (true);
   ui->editor_long_line_column->setValue (
     settings->value ("editor/long_line_column",80).toInt ());
   ui->editor_break_checkbox->setChecked (
     settings->value ("editor/break_lines",false).toBool ());
   ui->editor_break_checkbox->setChecked (
     settings->value ("editor/break_lines_comments",false).toBool ());
   ui->editor_wrap_checkbox->setChecked (
     settings->value ("editor/wrap_lines",false).toBool ());
@@ -887,20 +891,20 @@ settings_dialog::write_changed_settings 
   settings->setValue ("editor/showLineNumbers",
                       ui->editor_showLineNumbers->isChecked ());
   settings->setValue ("editor/line_numbers_size",
                       ui->editor_linenr_size->value ());
   settings->setValue ("editor/highlightCurrentLine",
                       ui->editor_highlightCurrentLine->isChecked ());
   settings->setValue ("editor/highlight_current_line_color",
                       _editor_current_line_color->color ());
-  settings->setValue ("editor/long_line_marker_none",
-                      ui->editor_long_line_marker_none->isChecked ());
   settings->setValue ("editor/long_line_marker",
                       ui->editor_long_line_marker->isChecked ());
+  settings->setValue ("editor/long_line_marker_line",
+                      ui->editor_long_line_marker_line->isChecked ());
   settings->setValue ("editor/long_line_marker_background",
                       ui->editor_long_line_marker_background->isChecked ());
   settings->setValue ("editor/long_line_column",
                       ui->editor_long_line_column->value ());
   settings->setValue ("editor/break_lines",
                       ui->editor_break_checkbox->isChecked ());
   settings->setValue ("editor/break_lines_comments",
                       ui->editor_break_comments_checkbox->isChecked ());
diff --git a/libgui/src/settings-dialog.ui b/libgui/src/settings-dialog.ui
--- a/libgui/src/settings-dialog.ui
+++ b/libgui/src/settings-dialog.ui
@@ -27,17 +27,17 @@
     <widget class="QTabWidget" name="tabWidget">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>0</height>
       </size>
      </property>
      <property name="currentIndex">
-      <number>8</number>
+      <number>1</number>
      </property>
      <widget class="QWidget" name="tab_general">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <attribute name="title">
        <string>General</string>
       </attribute>
@@ -47,18 +47,18 @@
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <widget class="QWidget" name="scrollAreaWidgetContents_2">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
-            <width>658</width>
-            <height>573</height>
+            <width>570</width>
+            <height>404</height>
            </rect>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_17">
            <item>
             <widget class="QGroupBox" name="groupBox">
              <property name="title">
               <string>Interface</string>
              </property>
@@ -1096,37 +1096,33 @@
                   <property name="text">
                    <string>Line length</string>
                   </property>
                  </widget>
                 </item>
                 <item row="0" column="5">
                  <layout class="QHBoxLayout" name="horizontalLayout_3">
                   <item>
-                   <widget class="QLabel" name="label_24">
+                   <widget class="QCheckBox" name="editor_long_line_marker">
                     <property name="text">
-                     <string>Long line marker:</string>
-                    </property>
-                   </widget>
-                  </item>
-                  <item>
-                   <widget class="QRadioButton" name="editor_long_line_marker_none">
-                    <property name="text">
-                     <string>None</string>
+                     <string>Long line marker</string>
                     </property>
                     <property name="checked">
                      <bool>true</bool>
                     </property>
                    </widget>
                   </item>
                   <item>
-                   <widget class="QRadioButton" name="editor_long_line_marker">
+                   <widget class="QRadioButton" name="editor_long_line_marker_line">
                     <property name="text">
                      <string>Line</string>
                     </property>
+                    <property name="checked">
+                     <bool>true</bool>
+                    </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QRadioButton" name="editor_long_line_marker_background">
                     <property name="text">
                      <string>Background</string>
                     </property>
                    </widget>
@@ -1815,18 +1811,18 @@
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <widget class="QWidget" name="scrollAreaWidgetContents_4">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
-            <width>658</width>
-            <height>573</height>
+            <width>481</width>
+            <height>230</height>
            </rect>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_7">
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_8">
              <item>
               <layout class="QGridLayout" name="gridLayout_7">
                <property name="sizeHint" stdset="0">
@@ -2096,18 +2092,18 @@
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <widget class="QWidget" name="scrollAreaWidgetContents_5">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
-            <width>658</width>
-            <height>573</height>
+            <width>474</width>
+            <height>196</height>
            </rect>
           </property>
           <layout class="QGridLayout" name="gridLayout_8">
            <item row="0" column="0">
             <widget class="QGroupBox" name="groupBox_4">
              <property name="title">
               <string>Behavior</string>
              </property>
@@ -2240,18 +2236,18 @@
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <widget class="QWidget" name="scrollAreaWidgetContents_6">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
-            <width>658</width>
-            <height>573</height>
+            <width>200</width>
+            <height>77</height>
            </rect>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_19">
            <item>
             <layout class="QGridLayout" name="gridLayout_14">
              <property name="topMargin">
               <number>0</number>
              </property>
@@ -2309,18 +2305,18 @@
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <widget class="QWidget" name="scrollAreaWidgetContents_8">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
-            <width>658</width>
-            <height>573</height>
+            <width>364</width>
+            <height>210</height>
            </rect>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_25">
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_26">
              <property name="sizeConstraint">
               <enum>QLayout::SetDefaultConstraint</enum>
              </property>
@@ -2508,18 +2504,18 @@
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <widget class="QWidget" name="scrollAreaWidgetContents_7">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
-            <width>658</width>
-            <height>573</height>
+            <width>529</width>
+            <height>199</height>
            </rect>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_20">
            <item>
             <layout class="QVBoxLayout" name="verticalLayout">
              <item>
               <widget class="QCheckBox" name="checkbox_allow_web_connect">
                <property name="text">
@@ -2667,18 +2663,18 @@
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <widget class="QWidget" name="scrollAreaWidgetContents_3">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
-            <width>658</width>
-            <height>573</height>
+            <width>615</width>
+            <height>169</height>
            </rect>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_vesc_manual">
            <property name="geometry" stdset="0">
             <rect>
              <x>0</x>
              <y>0</y>
              <width>678</width>
@@ -3440,10 +3436,42 @@
      <y>98</y>
     </hint>
     <hint type="destinationlabel">
      <x>429</x>
      <y>98</y>
     </hint>
    </hints>
   </connection>
+  <connection>
+   <sender>editor_long_line_marker</sender>
+   <signal>toggled(bool)</signal>
+   <receiver>editor_long_line_marker_line</receiver>
+   <slot>setEnabled(bool)</slot>
+   <hints>
+    <hint type="sourcelabel">
+     <x>252</x>
+     <y>397</y>
+    </hint>
+    <hint type="destinationlabel">
+     <x>346</x>
+     <y>397</y>
+    </hint>
+   </hints>
+  </connection>
+  <connection>
+   <sender>editor_long_line_marker</sender>
+   <signal>toggled(bool)</signal>
+   <receiver>editor_long_line_marker_background</receiver>
+   <slot>setEnabled(bool)</slot>
+   <hints>
+    <hint type="sourcelabel">
+     <x>252</x>
+     <y>397</y>
+    </hint>
+    <hint type="destinationlabel">
+     <x>428</x>
+     <y>397</y>
+    </hint>
+   </hints>
+  </connection>
  </connections>
 </ui>
